File: bench/ruby/primes.rb
n = (ENV["BENCH_N"] || "150000").to_i

def prime?(n)
  return false if n < 2
  limit = Integer.sqrt(n)   # exact integer sqrt, once — no d*d in the loop
  d = 2
  while d <= limit
    return false if n % d == 0
    d += 1
  end
  true
end

count = 0
(2...n).each { |k| count += 1 if prime?(k) }

puts count

File: bench/clojure/primes.clj
(defn prime? [^long n]
  (if (< n 2)
    false
    (loop [d 2]
      (cond (> (* d d) n) true
            (zero? (rem n d)) false
            :else (recur (inc d))))))
(let [n (long (Long/parseLong (or (System/getenv "BENCH_N") "150000")))]
  (println (count (filter prime? (range 2 n)))))
